gamemodes/terrortown/gamemode/server/sv_player.lua

```diff
@@ -511,88 +514,76 @@ function GM:KeyPress(ply, key)
-
-            return true
+    if ent.player_ragdoll then
+        if not ply:KeyDown(IN_WALK) then
+            CORPSE.ShowSearch(ply, ent)
         else
-            -- do nothing, can't +use held objects
-            return true
+            ply:Spectate(OBS_MODE_IN_EYE)
+            ply:SpectateEntity(ent)
         end
-    elseif tr.Entity.player_ragdoll then
-        CORPSE.ShowSearch(ply, tr.Entity, ply:KeyDown(IN_WALK) or ply:KeyDownLast(IN_WALK)) -- Body Corpse Search Identify TODO
-
-        return true
+    elseif ent:IsPlayer() and ent:IsActive() then
+        ply:Spectate(ply.spec_mode or OBS_MODE_IN_EYE)
+        ply:SpectateEntity(ent)
+    else
+        PROPSPEC.Target(ply, ent)
     end
 end
 
----
--- Normally all dead players are blocked from IN_USE on the server, meaning we
--- can't let them search bodies. This sucks because searching bodies is
--- fun. Hence on the client we override +use for specs and use this instead.
-local function SpecUseKey(ply, cmd, arg)
-    if not IsValid(ply) or not ply:IsSpec() then
+net.Receive("TTT2PlayerUseEntity", function(len, ply)
+    local ent = net.ReadEntity()
+
+    if not IsValid(ent) then
         return
     end
 
-    -- Do not allow the spectator to gather information if they're about to revive.
-    if ply:IsReviving() then
-        LANG.Msg(ply, "spec_about_to_revive", nil, MSG_MSTACK_WARN)
-
+    -- Check if the use interaction is possible
+    local distance = ply:GetShootPos():Distance(ent:GetPos())
+    if distance > 150 then
         return
     end
 
-    -- longer range than normal use
-    local tr = util.QuickTrace(ply:GetShootPos(), ply:GetAimVector() * 128, ply)
+    -- Override player use to enable addons to still handle it
+    -- Otherwise default to old IsTerror Check
+    overrideDoPlayerUse = ply:IsTerror()
+    if hook.Run("PlayerUse", ply, ent) then
+        ent:Use(ply, ply)
+    end
+    overrideDoPlayerUse = false
 
-    if not tr.Hit or not IsValid(tr.Entity) then
+    if ply:IsSpec() then
+        SpecUseKey(ply, ent)
         return
     end
 
-    if tr.Entity.player_ragdoll then
-        if not ply:KeyDown(IN_WALK) then
-            CORPSE.ShowSearch(ply, tr.Entity)
-        else
-            ply:Spectate(OBS_MODE_IN_EYE)
-            ply:SpectateEntity(tr.Entity)
+    if not ply:IsTerror() then
+        return
+    end
+
+    if ent.CanUseKey and ent.UseOverride then
+        local phys = ent:GetPhysicsObject()
+
+        if not IsValid(phys) or phys:HasGameFlag(FVPHYSICS_PLAYER_HELD) then
+            return
         end
-    elseif tr.Entity:IsPlayer() and tr.Entity:IsActive() then
-        ply:Spectate(ply.spec_mode or OBS_MODE_IN_EYE)
-        ply:SpectateEntity(tr.Entity)
-    else
-        PROPSPEC.Target(ply, tr.Entity)
+
+        ent:UseOverride(ply)
+    elseif ent.player_ragdoll then
+        CORPSE.ShowSearch(ply, ent, ply:KeyDown(IN_WALK) or ply:KeyDownLast(IN_WALK)) -- Body Corpse Search Identify TODO
```

Comment:
you should handle this todo

Reply:
Thats not mine 😅

Reply:
I think our policy is always to remove ToDos and put them in issues. But i think @Histalek is the expert with that

Reply:
Mhh, I can remove it, but I have no idea why its there in the first place, so I am clueless what I should put in the issue.

Reply:
```
$ git blame -L 552,552 gamemodes/terrortown/gamemode/server/sv_player.lua
ff244a6bd (Alf Müller 2019-10-29 03:27:54 +0100 552)         CORPSE.ShowSearch(ply, tr.Entity, ply:KeyDown(IN_WALK) or ply:KeyDownLast(IN_WALK)) -- Body Corpse Search Identify TODO
```

remove it tbh